content/docs/atom-guide/look-dev/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,669 @@
+   }
+   ```
+5. Repeat steps 3-4 with the depth pass in `o3de\Gems\Atom\Feature\Common\Assets\Materials\Types\VegetationBending_DepthPass.azsli` and the depth pass with PS in `o3de\Gems\Atom\Feature\Common\Assets\Materials\Types\VegetationBending_DepthPass_WithPS.azsli`.
+6. Open the **Editor** and you should see your tree swaying with the leaves still bending. If you don't, try opening the **Material Editor** and increasing the wind parameters.
+
+Amazing, our tree now sways and reacts to wind! Try to place multiple trees and observe how the trees sway differently when close together versus farther away. Also, add some lighting to make the trees pop!
+
+{{< video src="/images/atom-guide/vegetation-bending-tutorial/tree3.mp4" autoplay="true" loop="true" width="100%" muted="true" info="Video of trees swaying." >}}
+
+### Add motion vectors
+Because our tree is moving, we can add a *motion vector pass* so we can choose to add motion blur or other effects in the future.
+
+If you take a look at the `MeshMotionVectorVegetationBending.azsli` file, you can see that the vertex shader looks similar to the other vertex shaders we have seen, but there's a new output `OUT.m_worldPosPrev`. This is the position of the vector in the last time frame. The pixel shader uses both the previous vector position and the current one to calculate the motion vector. 
+
+However, we don't have an vertex shader input that gives us the previous position. Therefore, in the vertex shader, we want to calculate the bending for not only the current time as we have been, but also for the previous time frame.
+
+1. Download the three files in the `MeshMotionVectorVegetationBending` folder and move it to `o3de\Gems\Atom\Feature\Common\Assets\Materials\Types`.
+2. The `MeshMotionVectorVegetationBending` files already contain all the previous steps that we have done, like adding the shader option and calling the bending functions. 
+2. Open `VegetationBending.materialtype`. Add the bottom of the `shaders` list, add the motion vector pass:
+   
+   ```
+   {
+      "file": "./MeshMotionVectorVegetationBending.shader",
+      "tag": "MeshMotionVector"
+   }
+   ```
+3. Remember the `.lua` files and how we could enable/disable shaders? Let's ensure our motion vector shader is enabled. Open `VegetationBending_ShaderEnable.lua`.
+4. At the bottom of the `Process` fuunction, outside of any conditionals, enable the motion vector shader:
+   
+   ```
+   local motionPass = context:GetShaderByTag("MeshMotionVector")
+   motionPass:SetEnabled(true)
+   ```
+
+Great, now we have included the motion vector shader, so we have to go in and edit the vertex shader. Before that, however, we need to have a way to get the previous frame's time to use in our bending calculations. The `SceneSrg` has the current frame's time (`m_time`), but it doesn't contain the previous frame's time. Let's add the previous time in the `SceneSrg`.
```

Comment:
I think this is great that you are showing people how to add stuff to the SceneSrg. But we should try and teach people how to keep customizations in their game project while avoiding making changes to core renderer files as much as possible. The SceneSrg (and ViewSrg) are using a "partial" keyword. This means that any occurrence of "partial SceneSrg" will be merged with all others. You can add a new SceneSrg.azsli file to your game project and add anything you want to it there. Then add a SceneSrg.srgi file in {your-game-project-folder}\ShaderLib that includes your SceneSrg.azsli as well as SceneSrgAll.azsli. By doing it this way, you shouldn't have to touch core engine files at all. Search for existing "SceneSrg.srgi" and "SceneSrg.azsli" files to get a sense of how this works.

So your SceneSrg definition will only contain the new stuff you want to add like this:
```
partial ShaderResourceGroup SceneSrg
{
    float m_prevTime;
}
```

But the fact remains that you have to make changes in C++ to populate the data as well. There are ways to do this without touching existing code, but it would probably require making a new FeatureProcessor which is too much for this tutorial.

So actually as I'm thinking about this, I think we should just add your m_prevTime to the engine and remove this part of the tutorial. It seems like a very reasonable thing to have built-in.

Reply:
BTW, I do still think it would be great to teach people how to customize the SceneSrg properly _and_ add a new FeatureProcessor. We might consider writing a Part 2 of this tutorial where you implement a Wind Component that controls wind speed, direction, etc for an entire scene, instead of on a per-material basis. These wind settings would become new SceneSrg fields that the VegetationBending material type (and other wind-responsive material types) will pull from, instead of getting them from the MaterialSrg. Then (maybe in a Part 3) show users how to take the custom partial SceneSrg, custom material types, custom FeatureProcessor and put them all in a new Wind gem that could be shared by multiple game projects.

That might be out of scope for your time on this project, I'm  not sure, but I think this would be a great natural progression from your tutorial.

Reply:
Just to recap what was brought up offline: adding it to the SceneSrg would be great, but it doesn't work for all use cases, in particular when there are multiple render pipelines that update at different frequencies. If you have two render pipelines that are part of the same scene, and then both update in the same simulation frame, but one was last updated the previous simulation frame, and the other was last updated 3 simulation frames ago, they would have two different previous times, but would both be referring to the same SceneSrg.

For now, having part of the tutorial bet to update the previous time this way is fine, with the caveat that it wouldn't work with multiple pipelines updating at different frequencies. We can then iterate on this in the future to improve it.

Reply:
Per the new convos offline: made a PR to add the previous time in the SceneSRG but with a warning comment about unexpected results if using different frequencies.